modules/aws-eks-self-managed-node-groups/locals.tf

```diff
@@ -54,13 +59,13 @@ locals {
   ec2_principal     = "ec2.${var.context.aws_partition_dns_suffix}"
 
   # EKS Worker Managed Policies
-  eks_worker_policies = toset(concat([
+  eks_worker_policies = local.create_iam_role ? toset(concat([
```

Comment:
Does this throw a type mismatch error? `toset({})` vs `[]`? You might have to do:
```hcl
eks_worker_policies = {for k,v in toset(concat([
    "${local.policy_arn_prefix}/AmazonEKSWorkerNodePolicy",
    "${local.policy_arn_prefix}/AmazonEKS_CNI_Policy",
    "${local.policy_arn_prefix}/AmazonEC2ContainerRegistryReadOnly",
    "${local.policy_arn_prefix}/AmazonSSMManagedInstanceCore"],
    local.self_managed_node_group["additional_iam_policies"
  ])) : k => v if local.create_iam_role }
```